File: ZKCache/src/main/java/com/ericsson/ema/tim/dml/predicate/AbstractPredicate.java
package com.ericsson.ema.tim.dml.predicate;

import com.ericsson.ema.tim.dml.Clause;

public abstract class AbstractPredicate extends Clause {
    final String field;
    final Object valueToComp;

    AbstractPredicate(String field, Object value) {
        this.field = field;
        this.valueToComp = value;
    }

    @Override
    public String getField() {
        return field;
    }

    abstract public boolean eval(Object tuple);
}

File: zkcache_scala/src/main/scala/com/ericsson/ema/tim/dml/predicate/AbstractPredicate.scala
package com.ericsson.ema.tim.dml.predicate

import com.ericsson.ema.tim.dml.SelectClause

/**
  * Created by eqinson on 2017/5/12.
  */
abstract class AbstractPredicate(val field: String, val valueToComp: Object) extends SelectClause {
	override def getField: String = field

	def eval(tuple: Object): Boolean
}
